de-lib/src/main/java/org/iplantc/de/diskResource/client/presenters/details/DetailsViewPresenterImpl.java

```diff
@@ -26,49 +46,152 @@
     @Inject IplantAnnouncer announcer;
     @Inject FileSystemMetadataServiceFacade metadataService;
     @Inject DetailsView.Presenter.Appearance appearance;
+    @Inject
+    TagsServiceFacade tagsService;
+    @Inject
+    DiskResourceServiceFacade diskResourceService;
+    @Inject
+    SearchAutoBeanFactory sabFactory;
+
+    private List<Tag> tags = new ArrayList<>();
 
     private final DetailsView view;
 
+    private HandlerManager handlerManager;
+
+    IplantTagAutoBeanFactory factory;
+
     @Inject
-    DetailsViewPresenterImpl(final DetailsView view) {
+    DetailsViewPresenterImpl(final DetailsView view, final IplantTagAutoBeanFactory factory) {
         this.view = view;
+        this.factory = factory;
+
+        view.setPresenter(this);
+    }
 
-        view.addRemoveResourceTagSelectedHandler(this);
-        view.addUpdateResourceTagSelectedHandler(this);
+    @Override
+    public HandlerRegistration addSubmitDiskResourceQueryEventHandler(SubmitDiskResourceQueryEvent.SubmitDiskResourceQueryEventHandler handler) {
+        return ensureHandlers().addHandler(SubmitDiskResourceQueryEvent.TYPE, handler);
     }
 
     @Override
     public void onUpdateResourceTagSelected(UpdateResourceTagSelected event) {
         DiskResource resource = event.getDiskResource();
         Tag tag = event.getTag();
-        List<Tag> tagList = wrapInList(tag);
-        metadataService.attachTags(tagList, resource, new AsyncCallback<Void>() {
+        attachTag(tag.getId(), tag.getValue(), resource.getId(), null);
+
+    }
+
+    @Override
+    public void onRemoveResourceTagSelected(RemoveResourceTagSelected event) {
+        Tag tag = event.getTag();
+        DiskResource resource = event.getResource();
+        detachTag(tag.getId(), tag.getValue(), resource.getId(), null);
+    }
+
+    @Override
+    public void fetchTagsForResource(String diskResourceId, TagsFetchCallback callback) {
+        tags.clear();
+        metadataService.getTags(diskResourceId, new AsyncCallback<List<Tag>>() {
+            @Override
+            public void onFailure(Throwable caught) {
+                // FIXME Move to appearance
+                ErrorHandler.post("Unable to retrieve tags!", caught);
```

Comment:
This is probably a good time to move these strings over to the appearance like the comment says 😄